### PR TITLE
docs(Chip): update Chip docs w/ new formatting, new types

### LINKED ISSUE #20: Update template: Update Component.md template
Update https://github.com/reactioncommerce/catalyst/blob/10c7b998f091dbbf01a85f555afecd7c00da29bf/.reaction/scripts/templates/Component.md.template to fit our current needs

Remove Theme part

### PR DESCRIPTION
**Pull request by machikoyasuda**
Resolves #20 
Impact: **minor**  
Type: **docs|chore**

## Chip
- No Chip code changes
- Update docs formatting
- Added a new example type: Deletable chip

## Chore
- Update `Components.md.template`

## Breaking changes
None

### CODE REVIEW

**Comment by machikoyasuda** [comment]
Ready for review @rymorgan https://deploy-preview-51--heuristic-perlman-0ef024.netlify.com/#/Components/Content/Chip

I've made a future ticket to get the Typography for the Docs/Markdown and assigned you: https://github.com/reactioncommerce/catalyst/issues/53

File: package/src/components/Chip/Chip.md [hunk 1]
```diff
@@ -1,32 +1,50 @@
 ### Overview
 
-The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information. 
+Chips are compact visual elements that are used to present information to an operator to communicate a status or feedback on an action taken. Chips can be used to represent tags, badges, and other to convey compact bits of information to a user.
 
-### Usage
+[Chips](https://material.io/design/components/chips.html) allow users to enter information, make selections, filter content, or trigger actions.
 
-#### Default Catalyst chip
+The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information.
```
Comment (machikoyasuda) [inline package/src/components/Chip/Chip.md]:
```suggestion
The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information.
```

File: package/src/components/Chip/Chip.md [hunk 2]
```diff
@@ -1,32 +1,50 @@
 ### Overview
 
-The Catalyst Chip inherits from the Material-UI [Chip component](https://material-ui.com/components/chips/). Refer to the Material-UI [Chip API docs](https://material-ui.com/api/chip/) for more information. 
+Chips are compact visual elements that are used to present information to an operator to communicate a status or feedback on an action taken. Chips can be used to represent tags, badges, and other to convey compact bits of information to a user.
```
Comment (rymorgan) [inline package/src/components/Chip/Chip.md]:
```suggestion
Chips are compact visual elements that are used to present information to an operator to communicate status or feedback on an action taken. Chips can be used to represent tags, badges, and other compact bits of information to a user.
```

**Pull request merged**